### ISSUE TITLE
Broken hoobs server after recent update

### ISSUE THREAD

**Issue by Michael7911** [ebaauw/homebridge-hue#902]
After the recent update my hoobs service was stopped and could not be restarted. I tried a clean install of hoobs but as soon as I install homebridge-hue and try to configure my Philips Hue bridge and my Phoscon deCONZ bridge the service will stop and cannot be started again. 

I get the **[Hue] homebridge: error: Cannot add a bridged Accessory with the same UUID as another bridged Accessory:** error - which I did get once before some day but could resolve this issue - but now I can't. When trying ph discover it finds three bridges:

1) Phoscon on 127.0.0.1
2) Philips Hue
3) Phoscon on actual IP of hoobs server

Last thing I tried was to do a clean install and restore the backup - still I am not able to start the service. The log does not show any problems. I then updated the node but still no change. I tried a reboot and the server does not come up anymore - although ssh is still possible and Phoscon is also accessible.

What can I do? 

Many thanks and best regards!

**Comment by ebaauw** [comment]
You probably moved a device from the Hue bridge to the deCONZ gateway or the other way round, without deleting the resource from the old bridge/gateway.  Please attach the debug dump file, if you want me to check which device.

**Comment by Michael7911** [comment]
Thank you for your quick reply!

This is what I did now:
1) Cleaninstall of hoobs (flash Micro SD)
2) Create user and login to the server
3) General Settings (Language, Appearance, Celsius, etc.)
4) Update node with sudo npm install -g n and sudo n stable (Version is now 14.15.4)
5) Install deCONZ (I own a ConBee 2 Stick) with manual from hoobs (https://support.hoobs.org/docs/5ea6833b0ab68b0344e872e2)
6) Logged into deCONZ and connected it to my WIFI
7) In deCONZ I connected one of my Aqara Sensors and one of my OSRAM Lightify Plugs (both are not connected to Philipps Hue bridge, although I recently deleted the OSRAM Plug from Hue because I had it installed there)
Q: Do you know if some settings of the Phoscon Stick are stored on the stick itself? Can you reset this somehow?
8) Install homebridge-hue via hoobs
9) In Phoscon "Gateway", "Advanced" I clicked on "Connect App" to generate the user
10) I copied the user to the homebridge-hue config and saved the changes
Q: When I press "Save changes" a yellow line will move from left to right and back (I think you know what I mean) - this will not stop. When I refresh the page the settings are saved nevertheless. 
11) I pressed the button on the Philipps Hue bridge to generate the user
12) I copied the user to the homebridge-hue config and put a comma after the first entry. Now it looks like this:

"users": {
                "ABCDEFG12345": "ABCDEFG123",
                "HIJKLMNO6789": "HIJKLMNO678"
            },
12) When I entered the second user and clicked on "Save changes" everything worked fine - service stopped and started again. But the log shows again:

17.1.2021, 11:09:33 [Hue] Philips hue: request 28: POST / {"devicetype":"homebridge-hue#hoobs"}
17.1.2021, 11:09:33 [Hue] Philips hue: request 28: api error 101: link button not pressed
17.1.2021, 11:09:33 [Hue] Philips hue: press link button on the bridge to create a user - retrying in 15s

13) Although the user has been generated and saved to the config. So I pressed the button of the Hue bridge again and found this in the log:

"platforms": [
    {
      "platform": "Hue",
      "users": {
        "PQRSTUV12345678": "PQRSTUV987654"
      }
    }
  ]
17.1.2021, 11:14:33 [Hue] Philips hue: 1 accessories
17.1.2021, 11:14:33 [Hue] masked debug info dumped to /home/hoobs/.hoobs/etc/homebridge-hue.json.gz
17.1.2021, 11:14:33 Initializing platform accessory 'Phoscon-GW'...
17.1.2021, 11:14:33 [Hue] Phoscon-GW: 2 services
17.1.2021, 11:14:33 Initializing platform accessory 'Philips hue'...
17.1.2021, 11:14:33 [Hue] Philips hue: 2 services
17.1.2021, 11:14:33 Error: listen EADDRINUSE: address already in use :::51826
    at Server.setupListenHandle [as _listen2] (net.js:1318:16)
    at listenInCluster (net.js:1366:12)
    at Server.listen (net.js:1452:7)
    at EventedHTTPServer._this.listen (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/util/eventedhttp.js:100:30)
    at HAPServer._this.listen (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/HAPServer.js:201:31)
    at Bridge.Accessory._this.publish (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/Accessory.js:575:27)
    at Server.publishBridge (/usr/local/lib/node_modules/@hoobs/hoobs/bridge/server.js:193:21)
    at /usr/local/lib/node_modules/@hoobs/hoobs/bridge/server.js:162:53
    at async Server.start (/usr/local/lib/node_modules/@hoobs/hoobs/bridge/server.js:162:9)
17.1.2021, 11:14:33 Got SIGTERM, shutting down Bridge...
17.1.2021, 11:14:36 [Hue] goodbye
17.1.2021, 11:14:36 Got SIGINT, shutting down Bridge...
17.1.2021, 11:14:39 [Hue] goodbye

14) I copied the newly generated user to the homebridge-hue config and saved the changes.
15) Afterwards I checked the log and found this:

7.1.2021, 11:16:10 Loaded plugin 'homebridge-hue'
17.1.2021, 11:16:10 [1/17/2021, 3:16:10 AM] Registering platform 'homebridge-hue.Hue'
17.1.2021, 11:16:10 Loading 1 platforms...
17.1.2021, 11:16:10 [Hue] Initializing Hue platform...
17.1.2021, 11:16:10 [Hue] config.json: warning: plugin_map: ignoring unknown key
17.1.2021, 11:16:10 [Hue] config.json: warning: nupnp: ignoring unknown key
17.1.2021, 11:16:10 [Hue] homebridge-hue v0.12.17, node v14.15.4, homebridge v1.1.7
17.1.2021, 11:16:10 [Hue] searching bridges and gateways
17.1.2021, 11:16:15 [Hue] Phoscon-GW: dresden elektronik deCONZ gateway v2.7.1, api v1.16.0
17.1.2021, 11:16:15 [Hue] Philips hue: Signify Netherlands B.V. BSB002 bridge v1942135050, api v1.42.0
17.1.2021, 11:16:15 [Hue] Phoscon-GW: 1 accessories
17.1.2021, 11:16:15 [Hue] Philips hue: 1 accessories
17.1.2021, 11:16:15 [Hue] masked debug info dumped to /home/hoobs/.hoobs/etc/homebridge-hue.json.gz
17.1.2021, 11:16:15 Initializing platform accessory 'Phoscon-GW'...
17.1.2021, 11:16:15 [Hue] Phoscon-GW: 2 services
17.1.2021, 11:16:15 Initializing platform accessory 'Philips hue'...
17.1.2021, 11:16:15 [Hue] Philips hue: 2 services
17.1.2021, 11:16:15 Bridge is running on port 51826.

16) I went back to the config and changed "Expose Lights" to "Yes" saved the changes
17) Now all my Plugs showed up in Apple Home - the one connected to Phoscon and the others connected to Philipps Hue
18) I went back to the config and changed "Expose Sensors" to "Yes" and saved the changes
19) Now also my Aqara sensor showed up in Apple Home

This is a very long text, but I hope someone might find it useful. Although I still don't know what went wrong with my initial installation. I had configured many automations which I have to re-do now. This bothers me, but I'm also glad that it works now.

Many thanks and best regards.

**Comment by Michael7911** [comment]
It is me again. Unfortunately I was not as lucky as I hoped. I shut down the server to attach it to one of the OSRAM Plugs, so that I can remotely start it whenever I did an unintentional shut down.

Afterwards I tried to reboot it (Power on) but it does not come up. Phoscon is reachable and ssh also works. 
When I login with hoobs@hoobs.local I can start hoobs with "sudo hoobs" but then the User I created initially is gone and also the plugins I installed (tested this yesterday). So now I am sitting again on a server that does not come up.

Any suggestions?

**Comment by ebaauw** [comment]
I'm sorry, I don't know hoobs and cannot help you with that.  As mentioned in the README, I recommend to use a standard Homebridge installation.

> 17.1.2021, 11:14:33 Error: listen EADDRINUSE: address already in use :::51826

This probably means you're trying to start Homebridge while it's already running.  If Homebridge Hue asked for pressing the link button, and you're sure you entered the bridgeid/username in config.json correctly, then you're using a different config.json.

> Q: Do you know if some settings of the Phoscon Stick are stored on the stick itself? Can you reset this somehow?

Only the network parameters (pan id, network key, etc).  The configuration is stored in the REST API plugin database.  See "deCONZ for Dummies" on the REST API plugin Wiki.

> I shut down the server to attach it to one of the OSRAM Plugs, so that I can remotely start it whenever I did an unintentional shut down.

Isn't that a catch-22?  You need deCONZ and Homebridge to control the plug remotely, but you need to control the plug to start deCONZ and Homebridge?

**Comment by Michael7911** [comment]
Thank you. I hoped that you would know the hoobs server system as your software is listed as "Hoobs certified application"...

Anyway...I just did another cleaninstall of hoobs and restarted the server immediately and it worked. Then I only updated the node to 14.15.4 - as this is an error message I received from hombridge-hue when I installed I previously (Node is too old - update to 14.15.4). 

Afterwards I rebooted the server and it does not come up anymore. So clearly it has something to do with the node update what kills the server.

Do you have any suggestions what I can do? 

Meanwhile I will try Raspbian-homebridge-v1.0.17...

**Comment by ebaauw** [comment]
> Thank you. I hoped that you would know the hoobs server system as your software is listed as "Hoobs certified application"...

Not by me.

> Afterwards I rebooted the server and it does not come up anymore. So clearly it has something to do with the node update what kills the server.

> Do you have any suggestions what I can do?

Contact hoobs support?

> By the way: I have added this only plug for Raspberry-Pi to my Philipps Hue Bridge using "iConnectHue" app on iOS. With this app I can remotely start (Power on) the Pi.

Ah, cool.  It won't be visible in HomeKit, then.

**Comment by Michael7911** [comment]
In the last couple of days I have installed Raspbian Homebridge and re-installed all plugins and re-configured my automations. I am happy to say that everything works as of now. With Homebridge I was able to update the node to 14.15.4 without any issues. The server is stable and can be restarted without any problems. Therefore I will close this claim.

Thank you very much for your support and your work with this plugin!
